File: src/templates_cpp/src/publisher_template.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

class PublisherTemplate : public rclcpp::Node
{
    public:

    PublisherTemplate() : Node("publisher_template")
    {
        pub = this->create_publisher<std_msgs::msg::String>(
            "/example_topic",
            10
        );

        timer = this->create_wall_timer(
            std::chrono::milliseconds(1000),
            std::bind(&PublisherTemplate::publish, this)
        );
    }

    private:
    
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub;
    rclcpp::TimerBase::SharedPtr timer;

    void publish()
    {
        auto msg = std_msgs::msg::String();
        msg.data = "News";

        RCLCPP_INFO(this->get_logger(), "Transmitting: %s", msg.data.c_str());

        pub->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto publisher_template = std::make_shared<PublisherTemplate>();
    rclcpp::spin(publisher_template);
    rclcpp::shutdown();

    return 0;
}

File: src/templates_py/templates_py/publisher_template.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class PublisherTemplate(Node):

    def __init__(self, node_name: str) -> None:
        super().__init__(node_name)

        self.pub = self.create_publisher(
            msg_type=String,
            topic='/example_topic',
            qos_profile=10
        )

        self.timer = self.create_timer(
            timer_period_sec=1.0,
            callback=self.publish
        )

    def publish(self) -> None:
        msg = String()
        msg.data = 'Music'
        
        self.get_logger().info(message='Transmitting: ' + msg.data)

        self.pub.publish(msg=msg)

def main(args=None):
    rclpy.init(args=args)
    publisher_template = PublisherTemplate(node_name='publisher_template')
    rclpy.spin(node=publisher_template)
    rclpy.shutdown()

if __name__ == '__main__':
    main()